### ISSUE TITLE
`key` module: tracking issue for cryptographic key management support (e.g. list, generate, import)

### ISSUE THREAD

**Issue by tarcieri** [iqlusioninc/yubikey.rs#26]
This issue is for tracking work on the `key` module, which provides cryptographic key functionality.

If you are interested in testing or working on any of the functions below, please leave a comment in this issue.

Below are the functions in this module and their current status:

|       | Name                    | Testing                 | Description                               | Notes |
|----| -----------------| -----------------|------------------------------|-------|
| ✅ | `Key::list`            | `integration.rs`   | list keys                                      |            |
| ✅ | `Key::slot`            | manual                | get slot for key                          |            |
| ✅ | `Key::certificate` | manual                | get certificate for key              |            |
| ⚠️   | `generate`           | *none*                | generate key within device      | get rid of `GenerateKey`? |
| ⚠️   | `import`               | *none*                | import existing key                    | badly needs refactoring |
| ⚠️   | `attest`                 | *none*                | generate attestation cert         |            |
| ⚠️   | `sign_data`          | *none*                | generate digital signature        |            |
| ⚠️   | `decrypt_data`    | *none*                | decrypt data using key             |            |

**Comment by BlackHoleFox** [comment]
Evening. I was looking at refactoring the import functions, and likely after testing them with my YubiKey. I have some local work where I followed your comment on making a function for each key type, and I also did a fair bit of cleaning up around them too. 

Everything compiles and has a decent API, but I was curious about changing it so that for RSA keys, API users only need to provide `p` and `q`, and the optimization ones are generated inside the library. Would it be safe enough to adapt some of the code for calculating the other values from the `rsa` crate?
